src/node.rs

```diff
@@ -1,46 +1,14 @@
 use crate::raw::Table;
-use crossbeam_epoch::{Atomic, Guard, Shared};
+use crossbeam_epoch::Atomic;
 use parking_lot::Mutex;
-use std::borrow::Borrow;
-use std::sync::atomic::Ordering;
 
 /// Entry in a bin.
 ///
 /// Will _generally_ be `Node`. Any entry that is not first in the bin, will be a `Node`.
 #[derive(Debug)]
 pub(crate) enum BinEntry<K, V> {
     Node(Node<K, V>),
-    // safety: the pointer t to the next table inside Moved(t) is a valid pointer if the Moved(t)
-    // entry was read after loading `map::HashMap.table` while the guard used to load that table is
-    // still alive:
-    //
-    // When loading the current table of the HashMap with a guard g, the current epoch will be
-    // pinned by g. This happens _before_ the resize which put the Moved entry into the current
-    // table finishes, as otherwise a different table would have been loaded (see
-    // `map::HashMap::transfer`).
-    //
-    // Hence, for the Moved(t) read from the loaded table:
-    //
-    //   - When trying to access t during the current resize, t points to map::HashMap.next_table
-    //     and is thus valid.
-    //
-    //   - After the current resize and before another resize, `t == map::HashMap.table` as the
-    //     "next table" t pointed to during the resize has become the current table. Thus t is
-    //     still valid.
-    //
-    //   - The above is true until a subsequent resize ends, at which point `map::HashMap.table´ is
-    //     set to another new table != t and t is `epoch::Guard::defer_destroy`ed (again, see
-    //     `map::HashMap::transfer`). At this point, t is not referenced by the map anymore.
-    //     However, the guard g used to load the table is still pinning the epoch at the time of
-    //     the call to `defer_destroy`. Thus, t remains valid for at least the lifetime of g.
-    //
-    //   - After releasing g, either the current resize is finished and operations on the map
-    //     cannot access t anymore as a more recent table will be loaded as the current table
-    //     (see once again `map::HashMap::transfer`), or the argument is as above.
-    //
-    // Since finishing a resize is the only time a table is `defer_destroy`ed, the above covers
-    // all cases.
-    Moved(*const Table<K, V>),
+    Moved(),
```

Comment:
This should be just a marker variant now:
```suggestion
    Moved,
```